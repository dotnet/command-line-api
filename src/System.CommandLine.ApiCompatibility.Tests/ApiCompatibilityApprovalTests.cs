// Copyright (c) .NET Foundation and contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System.CommandLine.Hosting;
using System.CommandLine.NamingConventionBinder;
using System.CommandLine.Parsing;
using ApprovalTests;
using ApprovalTests.Reporters;
using Xunit;

namespace System.CommandLine.ApiCompatibility.Tests;

public class ApiCompatibilityApprovalTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void System_CommandLine_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContractForAssembly(typeof(ParseResult).Assembly);
        Approvals.Verify(contract);
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void System_CommandLine_Hosting_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContractForAssembly(typeof(HostingExtensions).Assembly);
        Approvals.Verify(contract);
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void System_CommandLine_NamingConventionBinder_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContractForAssembly(typeof(ModelBindingCommandHandler).Assembly);
        Approvals.Verify(contract);
    }
}