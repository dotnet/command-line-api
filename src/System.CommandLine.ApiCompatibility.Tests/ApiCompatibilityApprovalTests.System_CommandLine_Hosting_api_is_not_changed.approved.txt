System.CommandLine.Hosting
  public static class DirectiveConfigurationExtensions
    public static Microsoft.Extensions.Configuration.IConfigurationBuilder AddCommandLineDirectives(this Microsoft.Extensions.Configuration.IConfigurationBuilder config, System.CommandLine.ParseResult commandline, System.CommandLine.CliDirective directive)
  public static class HostingExtensions
    public static OptionsBuilder<TOptions> BindCommandLine<TOptions>(this OptionsBuilder<TOptions> optionsBuilder)
    public static Microsoft.Extensions.Hosting.IHost GetHost(this System.CommandLine.ParseResult parseResult)
    public static System.CommandLine.ParseResult GetParseResult(this Microsoft.Extensions.Hosting.IHostBuilder hostBuilder)
    public static System.CommandLine.ParseResult GetParseResult(this Microsoft.Extensions.Hosting.HostBuilderContext context)
    public static System.CommandLine.CliCommand UseCommandHandler<THandler>(this System.CommandLine.CliCommand command)
    public static System.CommandLine.CommandLineConfiguration UseHost(this System.CommandLine.CommandLineConfiguration builder, System.Action<Microsoft.Extensions.Hosting.IHostBuilder> configureHost = null)
    public static System.CommandLine.CommandLineConfiguration UseHost(this System.CommandLine.CommandLineConfiguration builder, System.Func<System.String[],Microsoft.Extensions.Hosting.IHostBuilder> hostBuilderFactory, System.Action<Microsoft.Extensions.Hosting.IHostBuilder> configureHost = null)
    public static Microsoft.Extensions.Hosting.IHostBuilder UseInvocationLifetime(this Microsoft.Extensions.Hosting.IHostBuilder host, System.Action<InvocationLifetimeOptions> configureOptions = null)
  public class InvocationLifetime, Microsoft.Extensions.Hosting.IHostLifetime
    .ctor(Microsoft.Extensions.Options.IOptions<InvocationLifetimeOptions> options, Microsoft.Extensions.Hosting.IHostEnvironment environment, Microsoft.Extensions.Hosting.IHostApplicationLifetime applicationLifetime, Microsoft.Extensions.Logging.ILoggerFactory loggerFactory = null)
    public Microsoft.Extensions.Hosting.IHostApplicationLifetime ApplicationLifetime { get; }
    public Microsoft.Extensions.Hosting.IHostEnvironment Environment { get; }
    public InvocationLifetimeOptions Options { get; }
    public System.Void Dispose()
    public System.Threading.Tasks.Task StopAsync(System.Threading.CancellationToken cancellationToken)
    public System.Threading.Tasks.Task WaitForStartAsync(System.Threading.CancellationToken cancellationToken)
  public class InvocationLifetimeOptions : Microsoft.Extensions.Hosting.ConsoleLifetimeOptions
    .ctor()
